src/hubdata/connect_target_data.py

```diff
@@ -0,0 +1,85 @@
+from pathlib import Path
+
+from pyarrow import dataset as ds
+from pyarrow import fs
+
+from hubdata import HubConnection, connect_hub
+from hubdata.create_target_data_schema import TargetType, create_target_data_schema
+
+
+def connect_target_data(hub_path: str | Path, target_type: TargetType) -> ds.dataset:
+    """
+    Top-level function for accessing the time-series target data or oracle-output target data for the passed `hub_path`.
+
```

Comment:
Can we also maybe add one or two examples of the call to help people understand the expected format?